src/exo/stdlib/halide_scheduling_ops.py

```diff
@@ -8,32 +8,63 @@
 from .inspection import *
 
 
-def get_affected_dim(proc, buffer_name: str, iter_sym) -> list[int]:
+def get_affected_dim(proc: Procedure, buffer_name: str, iter: str) -> list[int]:
     """
     Return which dimension of buffer are affected by [iter_sym]. Raises
     an error if there are multiple.
+
+    TODO: this should probably take a loop instead of an iter: str to scope the operation
     """
     dims = set()
     # TODO: this only matches against writes
-    for c in proc.find(f"{buffer_name}[_] = _", many=True):
+    for c in proc.find(f"{buffer_name} = _", many=True):
         for idx, idx_expr in enumerate(c.idx()):
             idx_vars = [
                 name.name() for (name, _) in get_reads_of_expr(idx_expr._impl._node)
             ]
-            if iter_sym in idx_vars:
+            if iter in idx_vars:
                 dims.add(idx)
 
     if len(dims) > 1:
-        raise ValueError(f"{iter_sym} affects multiple indices into {buffer_name}")
+        raise ValueError(f"{iter} affects multiple indices into {buffer_name}")
 
     return list(dims)[0]
 
 
+def _divide_with_recompute(
+    proc, loop_cursor: ForCursor, outer_hi, outer_stride: int, new_iters: list[str]
+):
+    """
+    Wrapper around `divide_with_recompute` which will not perform the operation
+    if it generates a trivial inner loop of size 1.
+    """
+    temp_proc = divide_with_recompute(
+        proc, loop_cursor, outer_hi, outer_stride, new_iters
+    )
+    temp_proc = simplify(temp_proc)
+    if is_literal(temp_proc, temp_proc.forward(loop_cursor).body()[0].hi(), 1):
+        return proc
+    return temp_proc
+
+
+# -------------------------------------------------------------#
+#                   Halide Scheduling Ops                      #
+# -------------------------------------------------------------#
+# TODO: add returning relevant cursors to all the scheduling ops
+
+
 def compute_at(proc: Procedure, producer_assign: AssignCursor, target_loop: ForCursor):
     """
     Computes the necessary values of producer at the level of [target_loop]
     in the consumer loop nest by fusing the two loop nests.
 
+    NOTE: This implementation assumes that both loop nests are at the top-level, but
+    it is easy to extend if we replace `get_toplevel_stmt` with an LCA function of
```

Comment:
There is a get_lca function in inspection.py if that helps.

Reply:
I think this is fine, mostly because Halide implicitly deals with top-level loop nests anyways with Funcs.